lib/kafo/parser_cache_reader.rb

```diff
@@ -1,24 +1,35 @@
 module Kafo
   class ParserCacheReader
-    def self.new_from_file(cache_path)
-      if cache_path.nil? || cache_path.empty?
-        logger.debug "No parser cache configured in :parser_cache_path, skipping setup"
+    def self.new_from_file(cache_paths)
+      cache_paths = [cache_paths].compact unless cache_paths.is_a?(Array)
+      if cache_paths.empty?
+        logger.debug "No parser cache(s) configured in :parser_cache_path, skipping setup"
         return nil
       end
 
-      unless File.exist?(cache_path)
-        logger.warn "Parser cache configured at #{cache_path} is missing, skipping setup"
+      non_existent = cache_paths.select { |path| !File.exist?(path) }
+      unless non_existent.empty?
+        logger.warn "Parser cache(s) configured at #{non_existent.join(", ")} are missing, skipping setup"
         return nil
       end
 
-      parsed = YAML.load(File.read(cache_path))
-      if !parsed.is_a?(Hash) || parsed[:version] != 1 || !parsed[:files].is_a?(Hash)
-        logger.warn "Parser cache is from a different version of Kafo, skipping setup"
-        return nil
+      parsed = cache_paths.map { |path| YAML.load(File.read(path)) }
+
+      parsed.each_with_index do |cache, i|
```

Comment:
`cache` is the parsed file, whereas i'm using `cache_paths[i]` to get the filepath to display incase of an error..